cmd/pipeline/deploy.go

```diff
@@ -0,0 +1,164 @@
+				name, err = getPipelineName()
+				if err != nil {
+					return err
+				}
+			}
+
+			if repository == "" {
+				return fmt.Errorf("repository is missing")
+			}
+
+			if namespace == "" {
+				namespace, err = getCurrentNamespace(ctx)
+				if err != nil {
+					return err
+				}
+			}
+
+			if err := deployPipeline(ctx, name, namespace, repository, branch, wait); err != nil {
+				return err
+			}
+
+			log.Success("Pipeline '%s' created", name)
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&name, "name", "p", "", "name of the pipeline (defaults to the folder name)")
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "namespace where the up command is executed (defaults to the current namespace)")
+	cmd.Flags().StringVarP(&repository, "repository", "r", "", "the repository to deploy (defaults to the current repository)")
+	cmd.Flags().StringVarP(&branch, "branch", "b", "", "the branch to deploy (defaults to the main branch)")
+	cmd.Flags().BoolVarP(&wait, "wait", "w", false, "wait until the pipeline finishes (defaults to false)")
+	return cmd
+}
+
+func deployPipeline(ctx context.Context, name, namespace, repository, branch string, wait bool) error {
+	spinner := utils.NewSpinner("Creating your pipeline...")
+	spinner.Start()
+	defer spinner.Stop()
+
+	_, err := okteto.DeployPipeline(ctx, name, namespace, repository, branch)
+	if err != nil {
+		return fmt.Errorf("failed to deploy pipeline: %w", err)
+	}
+
+	if !wait {
+		return nil
+	}
+
+	spinner.Update("Waiting for the pipeline to finish...")
+	return waitUntilRunning(ctx, name, namespace)
+}
+
+func getPipelineName() (string, error) {
+	workDir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Base(workDir), nil
+}
+
+func waitUntilRunning(ctx context.Context, name, namespace string) error {
+	t := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(5 * time.Minute)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("pipeline '%s' didn't finish after 5 minutes", name)
+		case <-t.C:
+			p, err := okteto.GetPipelineByName(ctx, name, namespace)
+			if err != nil {
+				if errors.IsNotFound(err) || errors.IsNotExist(err) {
+					return nil
+				}
+
+				return fmt.Errorf("failed to get pipeline '%s': %s", name, err)
+			}
+
+			switch p.Status {
+			case "running":
+				return nil
+			case "error":
+				return fmt.Errorf("pipeline '%s' failed", name)
+			default:
+				log.Infof("pipeline '%s' is '%s'", name, p.Status)
+			}
+		}
+	}
+}
+
+func getCurrentNamespace(ctx context.Context) (string, error) {
+	c, _, namespace, err := k8Client.GetLocal("")
+	if err != nil {
+		log.Infof("couldn't get the current namespace: %s", err)
+		return "", errors.UserError{
+			E:    fmt.Errorf("couldn't get the current namespace"),
+			Hint: "Run `okteto namespace`, or use the `--namespace` parameter",
+		}
+	}
+
+	ns, err := namespaces.Get(namespace, c)
+	if err != nil {
+		return namespace, nil
+	}
+
+	if !namespaces.IsOktetoNamespace(ns) {
```

Comment:
Instead of failing, I would use the default/personal Okteto Cloud namespace here (`space` not defined).
I could be pointing my context to minikube, for example, and run the pipeline on Okteto Cloud on a git hook.

Reply:
Although it could be confusing. Let's go with this for now.
I think we need the concept of okteto context sooner than later.

Reply:
yes. I think we could find a way of incorporate the context property of the okteto.yaml as part of this process (on a separate PR). But before that I want to focus on the flow and use cases this enable.